wsmaster/che-core-api-installer/src/main/java/org/eclipse/che/api/installer/server/InstallerRegistryService.java

```diff
@@ -66,76 +70,85 @@ public InstallerRegistryService(InstallerRegistry installerRegistry) {
-        }
+    public InstallerDto getInstaller(@ApiParam("The installer key") @PathParam("key") String key) throws InstallerException {
+        return asDto(installerRegistry.getInstaller(key));
     }
 
     @GET
     @Path("/{id}/versions")
     @Produces(APPLICATION_JSON)
     @ApiOperation(value = "Get a list of available versions of the specified installer", response = List.class)
     @ApiResponses({@ApiResponse(code = 200, message = "The response contains available versions of the specified installers"),
-                   @ApiResponse(code = 404, message = "Installer not found"),
                    @ApiResponse(code = 500, message = "Internal server error occurred")})
-    public List<String> getVersions(@ApiParam("The installer id") @PathParam("id") String id) throws NotFoundException,
-                                                                                                     ServerException {
-        try {
-            return installerRegistry.getVersions(id);
-        } catch (InstallerNotFoundException e) {
-            throw new NotFoundException(e.getMessage());
-        } catch (InstallerException e) {
-            throw new ServerException(e.getMessage(), e);
-        }
+    public List<String> getVersions(@ApiParam("The installer id") @PathParam("id") String id) throws InstallerException {
+        return installerRegistry.getVersions(id);
     }
 
     @GET
     @Produces(APPLICATION_JSON)
     @ApiOperation(value = "Get a collection of the available installers", response = Installer.class, responseContainer = "collection")
     @ApiResponses({@ApiResponse(code = 200, message = "The response contains collection of available installers"),
+                   @ApiResponse(code = 400, message = "Bad Request"),
                    @ApiResponse(code = 500, message = "Internal server error occurred")})
-    public Collection<InstallerDto> getInstallers() throws ServerException {
+    public Response getInstallers(@ApiParam(value = "Max items") @QueryParam("maxItems") @DefaultValue("30") int maxItems,
+                                  @ApiParam(value = "Skip count") @QueryParam("skipCount") @DefaultValue("0") int skipCount)
+            throws InstallerException, BadRequestException {
+
         try {
-            return installerRegistry.getInstallers()
-                                    .stream()
-                                    .map(DtoConverter::asDto)
-                                    .collect(Collectors.toList());
-        } catch (InstallerException e) {
-            throw new ServerException(e.getMessage(), e);
+            Page<? extends Installer> installers = installerRegistry.getInstallers(maxItems, skipCount);
+            return Response.ok()
+                           .entity(installers.getItems())
+                           .header("Link", createLinkHeader(installers))
+                           .header("TotalItemsCount", installers.getTotalItemsCount())
```

Comment:
1. Why do you need to know the exact count of the items here?
2. Custom headers usually follow `X-` prefix naming